src/controllers/controller.h

```diff
@@ -1,8 +1,12 @@
 #pragma once
 
+#include <unicode/ucnv.h>
+#include <unicode/unistr.h>
```

Comment:
can't help to notice that this looks like plain C headers (at least judging by the impl below). ICU also has a native C++ API afaict which is likely easier to use.

Reply:
Ok. Do you know how I can use them?

Reply:
Ah, just found the module list, apparently there is no C++ equivalent. https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/